Logic/core/indexer/index.py

```diff
@@ -149,30 +149,63 @@ def check_add_remove_is_correct(self):
             print('Add is incorrect, document')
             return
 
+        if not index_before_add[Indexes.STARS.value].__contains__('tim'):
+            index_before_add[Indexes.STARS.value].setdefault('tim', {})
+
         if (set(index_after_add[Indexes.STARS.value]['tim']).difference(set(index_before_add[Indexes.STARS.value]['tim']))
                 != {dummy_document['id']}):
             print('Add is incorrect, tim')
             return
 
+        if not index_before_add[Indexes.STARS.value].__contains__('henry'):
+            index_before_add[Indexes.STARS.value].setdefault('henry', {})
+
         if (set(index_after_add[Indexes.STARS.value]['henry']).difference(set(index_before_add[Indexes.STARS.value]['henry']))
                 != {dummy_document['id']}):
             print('Add is incorrect, henry')
             return
+
+        if not index_before_add[Indexes.GENRES.value].__contains__('drama'):
+            index_before_add[Indexes.GENRES.value].setdefault('drama', {})
+
         if (set(index_after_add[Indexes.GENRES.value]['drama']).difference(set(index_before_add[Indexes.GENRES.value]['drama']))
                 != {dummy_document['id']}):
             print('Add is incorrect, drama')
             return
 
+        if not index_before_add[Indexes.GENRES.value].__contains__('crime'):
+            index_before_add[Indexes.GENRES.value].setdefault('crime', {})
+
         if (set(index_after_add[Indexes.GENRES.value]['crime']).difference(set(index_before_add[Indexes.GENRES.value]['crime']))
                 != {dummy_document['id']}):
             print('Add is incorrect, crime')
             return
 
+        if not index_before_add[Indexes.SUMMARIES.value].__contains__('good'):
+            index_before_add[Indexes.SUMMARIES.value].setdefault('good', {})
+
         if (set(index_after_add[Indexes.SUMMARIES.value]['good']).difference(set(index_before_add[Indexes.SUMMARIES.value]['good']))
                 != {dummy_document['id']}):
             print('Add is incorrect, good')
             return
 
+        # Change the index_before_remove to its initial form if needed
+
+        if len(index_before_add[Indexes.STARS.value]['tim']) == 0:

```

Comment:
Also, these lines can be reduced by defining a function, for example named `delete_dummy_keys` or something like this, and then pass in `Indexes.STARS.value` , `henry` and others to it.